Enhancement absis ale power regulation - first

rainbow-solutions opened the pull request:
# Description

> The Arduino Pro Micro have some issues when +5V comes not from USB but from VCC pin. To be able to use the usb VIN pin I add 2 USB connectors to pass through the data connection, but on the same way inject the +5V from the ABSIS BUS to the USB connection. 
![grafik](https://github.com/user-attachments/assets/70dcde94-2eb0-45cf-9c4b-de9898d34dc9)

To get a stable +5V source I add a step down voltage regulator to have +5.1V from the +12V of the ABSIS bus.
![grafik](https://github.com/user-attachments/assets/e4773e0f-34e9-4c64-be4e-4ab7af036db8)
The Arduino is powered only over the USB port, the VCC pin is used only as output to provide power additional PCB's over the VCC pins.
The USB ports on the ABSIS board are used to pass through the data pins of the USB connection, and add a second power source on the VBUS +5V USB wire. 
To prevent backdrop from USB to ABSIS bus and inverted USB cables there are 2 diodes on both USB and ABSIS VBUS wire.

Addresses # (issue)

## Motivation and Context
> Enable the power source from the ABSIS bus and USB. Injecting the +5V from the ABSIS bus directly to the USB VBUS wire. This should solve the backdrop from ABSIS BUS to USB cable and faulty Arduino voltage regulator when using the RAW pin

## Type of change
- changed size of the ABSIS ALE board in one direction to fit USB and SMD parts
- 
![grafik](https://github.com/user-attachments/assets/9ddd7ed4-8270-40ba-b3d5-9c307dbb29c5)


# How Has This Been Tested?
> Not yet tested, waiting for approval of schematics.

# Screenshots:
<Paste screenshots of the WAS/IS state. Show us what it was before your changes, and show us what it is now.>

# Checklist: (Delete non-relevant sections)

## MCAD:
 - the PCB is 5mm longer than before, there should be no impact on mounting positions, but needs to be checked.


## PCB/ECAD:
- waiting for approval of the schematics and PCB to continue on master parts and creating gerber / production files.

rainbow-solutions committed:
Add 12V USB injection

Changed PCB size and add USB injection of 12V from ABSIS bus

rainbow-solutions committed:
ABSIS bus +5V USB injection

Change PCB size and add USB injection of +12V -> +5V to power Arduino Pro Micro

rainbow-solutions committed:
C5

Add 2 missing connections

jrsteensen reviewed (changes requested):
Unfortunately, this isn't quite what we were looking for. We don't want to add USB cables just for power - that fix will come from the good OH pro-micros that don't have the power trace spacing issues all other pro-micros currently may or may not have, and some additional protections to prevent ABSIS bus power being injected on the USB power bus. If you can update to remove the USB connectors, that would be appreciated.

rainbow-solutions committed:
revert to version without usb

deleted both usb connectors and starting pcb design

rainbow-solutions committed:
Version with schematic & first layout

rainbow-solutions committed:
redraw zones

rainbow-solutions commented:
First version with power regulation without the USB connectors. The D1 and F1 now protects the raw pin and the entire power regulator. depending on what solder bridges are closed it will powered directly from the JP3 to the raw pin, or closing JP1 and CP2 over the +5V pin.
<img width="654" height="977" alt="image" src="https://github.com/user-attachments/assets/87298176-60d9-4af4-a956-013407e00004" />
The schematic is here:
<img width="755" height="262" alt="image" src="https://github.com/user-attachments/assets/4261e4b9-9a7a-465c-877d-d0e4695816c1" />

jrsteensen commented:
Sorry, just saw your additional commits while I was writing my reply. 😂

rainbow-solutions commented:
Oh no, you are right, I have to add the jumper description... :-) just a moment, I will add it right now...

rainbow-solutions committed:
add jumper description

rainbow-solutions committed:
line on the silkscreen removed...

kbastronomics commented:
I'm implementing a power switching circuit into the pro Micro that will favor the 12 V input or actually the raw input over the 5 V USB power and only allow one of them through. They'll be no direct connection to the regulator except through the raw Pin and the output from it will feed the VCC input of the 32U4.

That should solve most of the problems at the source, which is the power circuit of the pro Micro, which was never designed to allow both USB power and external power at the same time

jrsteensen commented:
This is a little late I know, after all the work @rainbow-solutions has put in, but is a change even necessary for the ALE? Will it all be fixed by the OH Pro Micro?

kbastronomics commented:
i don't think it will hurt,   I would have to program the 2121 to prioritize the externa, power whether it is 5v or 12V coming in on RAW/VCC in addition to the vBUS but honestly I think I would just block VCC from hitting the input of the regulator and only allow RAW or VBUS

kbastronomics commented:
I'm referring to a TPS2121 Power Mux
https://www.ti.com/product/TPS2121
it allows you to have 2 power sources and auto switch between based on a few different methods.  
in this case I'm looking to make it 12v priority which means if only 5v is available pass it on, if only 12v is available pass it on
but if both 5v and 12 v are available it will select 12v since I make it the priority power supply

There's like no space on the top side of the PCB so I'll have to put it on the bottom side, I'm also removing the SJ1 to remove that feeding of the VCC power back to the input of the regulator (mux in my case.

the only thing this does not address is the phantom power on the D+ of the USB port but that really is not an issue, its leakage current due to the firmware config not turning off the input pullup resistor on the 32u4 but we've talking <3v @ <1mA of leakage
its not blowing any hub

rainbow-solutions committed:
generate production files

generate production files with new compnents and update bom and placing file. change also value of fuse.

rainbow-solutions committed:
changed type of BOM and CPL file due jlcpcb import error

jrsteensen commented on ECAD/PCBs/ABSIS/ABSIS_ALE/MCAD/abteilung.stl:
This file should be a STEP file, not a STL.

rainbow-solutions replied:
Sorry, but I did not understand how can I create this step file, I try to look at the documentation but found nothing... sorry...

jrsteensen commented on ECAD/PCBs/ABSIS/ABSIS_ALE/JLCPCB/production_files/BOM-ABSIS_ALE.csv:
How are you doing your BOM generation? (Columns changed significantly.) 

WAS: Comment,Designator,Footprint,LCSC
IS: "Reference","Value","Datasheet","Footprint","Qty","DNP","LCSC"

jrsteensen reviewed (changes requested):
Please see individual file comments.

rainbow-solutions committed:
correct solder indication on silkscreen

rainbow-solutions committed:
updated bom file

rainbow-solutions committed:
create new step file

jrsteensen commented:
### SCHEMATIC CHANGES

- [x] F1: Change to 500mA (C7472569)
- [x] ADD SHOTTKEY DIODE BETWEEN JP2 & VCC (Same as D1)
- [x] Change U1 to use DIP package (MAX487EEPA+)
- [x] Change U1 to use VCC instead of 5V, remove 5V label.
- [x] Change Vin(+12V) to 12V_SUPPLY
- [x] Rewrite power note
- [x] Fix all values to be horizontal.

### LAYOUT CHANGES

- [ ] Change U1 to use DIP package (MAX487EEPA+)
- [ ] ADD SHOTTKEY DIODE BETWEEN JP2 & VCC (Same as D1)
- [ ] Silkscreen fixes.
- [ ] Optimize new circuit layout on PCB.

Got this in progress.

jrsteensen committed:
Clarify jumper instructions and rework nets

Replace the solder-jumper note with explicit, safety-focused instructions for JP1/JP2/JP3 and reposition that text. Large tidy-up of schematic nets: many junctions, wires and labels were repositioned or reconnected (UUIDs updated) to correct routing and visual layout. Update component U1 properties: change value/part to MAX487EEPA+, set DIP-8 footprint and move related properties/instances. Remove/adjust redundant power symbol entries, update power/ground label positions, and tweak JP3 placement and properties. Overall this commit is a schematic cleanup and clarification to improve documentation and net topology. Added D2 between regulator output and VCC.
